### PR TITLE
Feat/web3 add reason for vote

### PR DESCRIPTION
**Pull request by conradocanasm0**
Add optional reason for voting in standard proposals.

image

https://github.com/user-attachments/assets/0e0889eb-8a9b-426a-ab09-4bc9de5f360c

### CODE REVIEW

**Comment by conradocanasm0** [comment]
Usage of voting:

https://github.com/user-attachments/assets/0134b08d-f18b-4954-a004-8d5bf8cd9f83

File: pages/proposals/index.vue [hunk 1]
```diff
@@ -210,17 +221,47 @@ async function onCastBatchVotes() {
...
-      functionName: "castVotes",
-      args: [proposalIds, votes], // uint256 proposalId, uint8 support
-      account: userAccount.value,
-    });
+    const isOnlyOneVote = selectedCastProposals.value.length === 1;
+    const anyProposalHasReason = selectedCastProposals.value.some(
+      (p) => p.reason,
+    );
+
+    if (anyProposalHasReason && isOnlyOneVote) {
```
Comment (conradocanasm0) [inline pages/proposals/index.vue:236]: this can be better written but the most readable, wanting comments

File: components/proposal/Card.vue [hunk 2]
```diff
@@ -214,11 +255,8 @@ function onSingleCastSelected(vote: boolean) {
 
 const proposalId = computed(() => props.proposal.proposalId);
 const governor = computed(() => useGovernor({ proposalId: proposalId.value }));
-const pastEpoch = computed(() => BigInt(props.proposal.voteStart - 1));
 
 const { power: powerVotingPower } = useMVotingPower(userAccount);
-const { power: powerPastVotingPower, zero: zeroPastVotingPower } =
```
Comment (ernaneluis) [inline components/proposal/Card.vue:220]: why this was removed?

**Review by ernaneluis — CHANGES_REQUESTED** [review]

File: pages/proposals/index.vue (hunk 1 above)
Comment (ernaneluis) [reply pages/proposals/index.vue:236]: maybe wrap the content of each if into a function just for better readability

Comment (ernaneluis) [reply pages/proposals/index.vue:236]: hash = await castVote(...all stuf)

casVote implementation the ifs

castVotes(){
if() castVoteWithReason
else if() castVote
...

}

File: components/proposal/Card.vue (hunk 2 above)
Comment (conradocanasm0) [reply components/proposal/Card.vue:220]: it was shown as not used

**Review by ernaneluis — APPROVED** [review]

**Pull request merged**